packages/waku/src/router/client.ts

```diff
@@ -464,14 +467,49 @@ const handleScroll = () => {
   });
 };
 
+const overridePathAndQuery = async (
```

Comment:
This approach fails the use case of `unstable_rerenderRoute` which wasn't caught by tests.
#1406 is the fix for this, reverting the change in this PR.